### PR TITLE
Update sampler types

### LINKED ISSUE #113: Update sampler types
See https://github.com/open-telemetry/opentelemetry-specification/pull/609 for changes.

### PR DESCRIPTION
**Pull request by siyopao**
I tried working on this. I think I understood what needed to be done, but please let me know if I made a mistake. I didn't update the `tracing-grpc` example as it has some dependency conflicts with `tracing-opentelemetry`.

Resolves #113.

### CODE REVIEW

File: src/sdk/trace/sampler.rs [hunk 1]
```diff
@@ -5,11 +5,11 @@ use crate::api;
...
 pub enum Sampler {
     /// Always sample the trace
-    Always,
+    AlwaysOn,
     /// Never sample the trace
-    Never,
+    AlwaysOff,
     /// Sample if the parent span is sampled
-    Parent,
+    ParentOrElse,
```
Comment (jtescher) [inline src/sdk/trace/sampler.rs:12]: From looking at the spec this should be a composite sampler, so likely something like `ParentOrElse(Box<Sampler>)`, and the logic should reflect the inner sampler when there is no parent.

Comment (siyopao) [reply src/sdk/trace/sampler.rs]: Thank you for your comment. Sorry, yes, I reread the spec and I now think I am following it 😄  Please let me know if my changes are on the right track. Thanks!

File: src/sdk/trace/sampler.rs [hunk 2]
```diff
@@ -28,17 +28,29 @@ impl api::Sampler for Sampler {
...
-                }
-            }
+            Sampler::AlwaysOff => api::SamplingDecision::NotRecord,
+            // The parent decision if sampled; otherwise the decision of delegate_sampler
+            Sampler::ParentOrElse(delegate_sampler) => parent_context.map_or(
+                delegate_sampler
+                    .should_sample(
+                        parent_context,
+                        trace_id,
+                        _name,
```
Comment (jtescher) [inline src/sdk/trace/sampler.rs:40]: can we remove these underscores now that the variables are being used?

Comment (siyopao) [reply src/sdk/trace/sampler.rs]: 😬 Fixed

**Pull request merged**